{"bomFormat": "CycloneDX", "specVersion": "1.4", "serialNumber": "urn:uuid:97deb054-d8ca-4808-95ca-cfcc93810b9b", "version": 1, "metadata": {"timestamp": "2023-01-23T08:35:41.853Z", "tools": [{"vendor": "AppThreat", "name": "cdxgen", "version": "6.0.14"}], "authors": [{"name": "Team AppThreat", "email": "cloud@appthreat.com"}]}, "components": [{"publisher": "", "group": "", "name": "postgres", "version": "latest", "description": "", "licenses": [], "purl": "pkg:oci/postgres@latest?", "type": "library", "bom-ref": "pkg:oci/postgres@latest?", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "postgres"}]}, {"publisher": "", "group": "", "name": "mozilla/tls-observatory", "version": "latest", "description": "", "licenses": [], "purl": "pkg:oci/mozilla/tls-observatory@latest?", "type": "library", "bom-ref": "pkg:oci/mozilla/tls-observatory@latest?", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "github.com/google", "name": "pprof", "version": "v0.0.0-20200229191704-1ebb73c60ed3", "description": "", "hashes": [{"alg": "SHA-256", "content": "6605513e852afe17eacc0aa1eec1cca9bdc8f51ab90b9f5d233d926c1c09e1e3"}], "licenses": [], "purl": "pkg:golang/github.com/google/pprof@v0.0.0-20200229191704-1ebb73c60ed3", "type": "library", "bom-ref": "pkg:golang/github.com/google/pprof@v0.0.0-20200229191704-1ebb73c60ed3", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "github.com/ianlancetaylor", "name": "demangle", "version": "v0.0.0-20200414190113-039b1ae3a340", "description": "", "hashes": [{"alg": "SHA-256", "content": "6924af6ffb7a93598fa310ea3b8bc987a54e0863f05383b40b6fc4aa7761d527"}], "licenses": [], "purl": "pkg:golang/github.com/ianlancetaylor/demangle@v0.0.0-20200414190113-039b1ae3a340", "type": "library", "bom-ref": "pkg:golang/github.com/ianlancetaylor/demangle@v0.0.0-20200414190113-039b1ae3a340", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "arch", "version": "v0.0.0-20200511175325-f7c78586839d", "description": "", "hashes": [{"alg": "SHA-256", "content": "7e521a108e8b354eb13820f93da26f9fdc063f46a09883aa8ebb6c286460bafe"}], "licenses": [], "purl": "pkg:golang/golang.org/x/arch@v0.0.0-20200511175325-f7c78586839d", "type": "library", "bom-ref": "pkg:golang/golang.org/x/arch@v0.0.0-20200511175325-f7c78586839d", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "crypto", "version": "v0.0.0-20200622213623-75b288015ac9", "description": "", "hashes": [{"alg": "SHA-256", "content": "2f320f3107f230d86118f8549183ace4aa4be14f2b2ca7a65f5c862e10e052da"}], "licenses": [], "purl": "pkg:golang/golang.org/x/crypto@v0.0.0-20200622213623-75b288015ac9", "type": "library", "bom-ref": "pkg:golang/golang.org/x/crypto@v0.0.0-20200622213623-75b288015ac9", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "mod", "version": "v0.3.0", "description": "", "hashes": [{"alg": "SHA-256", "content": "b3442c8f5002b7d78fa7f84cca93377e5e1f66a444589c1d60312a8518996540"}], "licenses": [], "purl": "pkg:golang/golang.org/x/mod@v0.3.0", "type": "library", "bom-ref": "pkg:golang/golang.org/x/mod@v0.3.0", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "sys", "version": "v0.0.0-20200501145240-bc7a7d42d5c3", "description": "", "hashes": [{"alg": "SHA-256", "content": "87536359c7bd5d12c6404b16ef0a4a3428c6f43b4d942955b852c465d0cd6c4b"}], "licenses": [], "purl": "pkg:golang/golang.org/x/sys@v0.0.0-20200501145240-bc7a7d42d5c3", "type": "library", "bom-ref": "pkg:golang/golang.org/x/sys@v0.0.0-20200501145240-bc7a7d42d5c3", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "tools", "version": "v0.0.0-20200616133436-c1934b75d054", "description": "", "hashes": [{"alg": "SHA-256", "content": "124558419a00b18e39faba1892f818908878c61fea8e050af53758d974fde061"}], "licenses": [], "purl": "pkg:golang/golang.org/x/tools@v0.0.0-20200616133436-c1934b75d054", "type": "library", "bom-ref": "pkg:golang/golang.org/x/tools@v0.0.0-20200616133436-c1934b75d054", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "xerrors", "version": "v0.0.0-20200806184451-1a77d5e9f316", "description": "", "hashes": [{"alg": "SHA-256", "content": "23fe73ebdf2c9fd29af1378973d30aae85147ea0416ae5a342a2c9d8e3df998d"}], "licenses": [], "purl": "pkg:golang/golang.org/x/xerrors@v0.0.0-20200806184451-1a77d5e9f316", "type": "library", "bom-ref": "pkg:golang/golang.org/x/xerrors@v0.0.0-20200806184451-1a77d5e9f316", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "net", "version": "v0.0.0-20210428183841-261fb518b1ed", "description": "", "hashes": [{"alg": "SHA-256", "content": "fceed5d30680f2bedc80687cd51a37a3584ec6ddf648c54f89c66ba0a436b190"}], "licenses": [], "purl": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "type": "library", "bom-ref": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "", "group": "golang.org/x", "name": "text", "version": "v0.3.3-0.20200430171850-afb9336c4530", "description": "", "hashes": [{"alg": "SHA-256", "content": "e59a1cfd046d2955b34213ad04cbea1f30e917a8ab3bdcfdf310dc7a8b2e1a24"}], "licenses": [], "purl": "pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "type": "library", "bom-ref": "pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "Steven Bethard", "group": "", "name": "argparse", "version": "1.2.1", "description": "Python command-line parsing library", "licenses": [], "purl": "pkg:pypi/argparse@1.2.1", "type": "library", "bom-ref": "pkg:pypi/argparse@1.2.1", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "Matt Mackall and many others", "group": "", "name": "mercurial", "version": "4.8.2", "description": "Fast scalable distributed SCM (revision control, version control) system", "licenses": [], "purl": "pkg:pypi/mercurial@4.8.2", "externalReferences": [{"type": "website", "url": "https://mercurial-scm.org/"}], "type": "library", "bom-ref": "pkg:pypi/mercurial@4.8.2", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "Guido van Rossum and the Python community", "group": "", "name": "Python", "version": "2.7.16", "description": "A high-level object-oriented programming language", "licenses": [], "purl": "pkg:pypi/python@2.7.16", "externalReferences": [{"type": "website", "url": "http://www.python.org/2.7"}], "type": "library", "bom-ref": "pkg:pypi/python@2.7.16", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}, {"publisher": "Phillip J. Eby", "group": "", "name": "wsgiref", "version": "0.1.2", "description": "WSGI (PEP 333) Reference Library", "licenses": [], "purl": "pkg:pypi/wsgiref@0.1.2", "type": "library", "bom-ref": "pkg:pypi/wsgiref@0.1.2", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "oci:SrcImage", "value": "mozilla/tls-observatory"}]}], "services": [{"bom-ref": "urn:service:postgres:latest", "name": "postgres", "version": "latest", "group": "", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "ServiceName", "value": "postgres"}]}, {"bom-ref": "urn:service:api:latest", "name": "api", "version": "latest", "group": "", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/deployment.yaml"}, {"name": "ServiceName", "value": "api"}]}, {"bom-ref": "urn:service:setup-db:latest", "name": "setup-db", "version": "latest", "group": "", "properties": [{"name": "SrcFile", "value": "/github/workspace/kubernetes/setup_db.yaml"}, {"name": "ServiceName", "value": "setup-db"}]}], "dependencies": [], "vulnerabilities": [{"bom-ref": "CVE-2021-38561/pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "id": "CVE-2021-38561", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-38561"}, "ratings": [{"score": 2.0, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nDue to improper index calculation, an incorrectly formatted language tag can cause Parse to panic via an out of bounds read. If Parse is used to process untrusted user inputs, this may be used as a vector for a denial of service attack.\n\n            ", "recommendation": "Update to 0.3.8 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "versions": [{"version": "v0.3.3-0.20200430171850-afb9336c4530", "status": "affected"}, {"version": "0.3.8", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2021-33194/pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "id": "CVE-2021-33194", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-33194"}, "ratings": [{"score": 2.0, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nAn attacker can craft an input to ParseFragment that causes it to enter an infinite loop and never return.\n\n            ", "recommendation": "Update to 0.4.0 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "versions": [{"version": "v0.0.0-20210428183841-261fb518b1ed", "status": "affected"}, {"version": "0.4.0", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2021-44716/pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "id": "CVE-2021-44716", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-44716"}, "ratings": [{"score": 3.7, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nAn attacker can cause unbounded memory growth in servers accepting HTTP/2 requests.\n\n            ", "recommendation": "Update to 0.4.0 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "versions": [{"version": "v0.0.0-20210428183841-261fb518b1ed", "status": "affected"}, {"version": "0.4.0", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-27664/pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "id": "CVE-2022-27664", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-27664"}, "ratings": [{"score": 3.7, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nHTTP/2 server connections can hang forever waiting for a clean shutdown that was preempted by a fatal error. This condition can be exploited by a malicious client to cause a denial of service.\n\n            ", "recommendation": "Update to 0.4.0 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "versions": [{"version": "v0.0.0-20210428183841-261fb518b1ed", "status": "affected"}, {"version": "0.4.0", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-32149/pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "id": "CVE-2022-32149", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-32149"}, "ratings": [{"score": 2.0, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nAn attacker may cause a denial of service by crafting an Accept-Language header which ParseAcceptLanguage will take significant time to parse.\n\n## Related CVE(s)\nCVE-2022-32149, GHSA-69ch-w2m2-3vjp\n            \n            ", "recommendation": "Update to 0.3.8 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/text@v0.3.3-0.20200430171850-afb9336c4530", "versions": [{"version": "v0.3.3-0.20200430171850-afb9336c4530", "status": "affected"}, {"version": "0.3.8", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-41717/pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "id": "CVE-2022-41717", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-41717"}, "ratings": [{"score": 3.7, "severity": "low", "method": "CVSSv31"}], "cwes": [], "description": "# Summary\nAn attacker can cause excessive memory growth in a Go server accepting HTTP/2 requests.\n\nHTTP/2 server connections contain a cache of HTTP header keys sent by the client. While the total number of entries in this cache is capped, an attacker sending very large keys can cause the server to allocate approximately 64 MiB per open connection.\n\n            ", "recommendation": "Update to 0.4.0 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:golang/golang.org/x/net@v0.0.0-20210428183841-261fb518b1ed", "versions": [{"version": "v0.0.0-20210428183841-261fb518b1ed", "status": "affected"}, {"version": "0.4.0", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-0391/pkg:pypi/python@2.7.16", "id": "CVE-2022-0391", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-0391"}, "ratings": [{"score": 7.5, "severity": "high", "method": "CVSSv31"}], "cwes": [74], "description": "A flaw was found in Python, specifically within the urllib.parse module. This module helps break Uniform Resource Locator (URL) strings into components. The issue involves how the urlparse method does not sanitize input and allows characters like '\\r' and '\\n' in the URL path. This flaw allows an attacker to input a crafted URL, leading to injection attacks. This flaw affects Python versions prior to 3.10.0b1, 3.9.5, 3.8.11, 3.7.11 and 3.6.14.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "Mailing List", "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UDBDBAU6HUPZHISBOARTXZ5GKHF2VH5U/"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpuapr2022.html"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-26488/pkg:pypi/python@2.7.16", "id": "CVE-2022-26488", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-26488"}, "ratings": [{"score": 7.0, "severity": "high", "method": "CVSSv31"}], "cwes": [426], "description": "In Python before 3.10.3 on Windows, local users can gain privileges because the search path is inadequately secured. The installer may allow a local attacker to add user-writable directories to the system search path. To exploit, an administrator must have installed Python for all users and enabled PATH entries. A non-administrative user can trigger a repair that incorrectly adds user-writable paths into PATH, enabling search-path hijacking of other users and system services. This affects Python (CPython) through 3.7.12, 3.8.x through 3.8.12, 3.9.x through 3.9.10, and 3.10.x through 3.10.2.", "recommendation": "Update to 3.6.14 or later", "advisories": [], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": ""}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2022-45061/pkg:pypi/python@2.7.16", "id": "CVE-2022-45061", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2022-45061"}, "ratings": [{"score": 7.5, "severity": "high", "method": "CVSSv31"}], "cwes": [400], "description": "An issue was discovered in Python before 3.11.1. An unnecessary quadratic algorithm exists in one path when processing some inputs to the IDNA (RFC 3490) decoder, such that a crafted, unreasonably long name being presented to the decoder could lead to a CPU denial of service. Hostnames are often supplied by remote servers that could be controlled by a malicious actor; in such a scenario, they could trigger excessive CPU consumption on the client attempting to make use of an attacker-supplied supposed hostname. For example, the attack payload could be placed in the Location header of an HTTP response with status code 302. A fix is planned in 3.11.1, 3.10.9, 3.9.16, 3.8.16, and 3.7.16.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "GitHub Issue", "url": "https://github.com/python/cpython/issues/98433"}, {"title": "Mailing List", "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7WQPHKGNXUJC3TC3BDW5RKGROWRJVSFR/"}, {"title": "vendor", "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7WQPHKGNXUJC3TC3BDW5RKGROWRJVSFR/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2021-23336/pkg:pypi/python@2.7.16", "id": "CVE-2021-23336", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-23336"}, "ratings": [{"score": 5.9, "severity": "medium", "method": "CVSSv31"}], "cwes": [444], "description": "The package python/cpython from 0 and before 3.6.13, from 3.7.0 and before 3.7.10, from 3.8.0 and before 3.8.8, from 3.9.0 and before 3.9.2 are vulnerable to Web Cache Poisoning via urllib.parse.parse_qsl and urllib.parse.parse_qs by using a vector called parameter cloaking. When the attacker can separate query parameters using a semicolon (;), they can cause a difference in the interpretation of the request between the proxy (running with default configuration) and the server. This can result in malicious requests being cached as completely safe ones, as the proxy would usually not see the semicolon as a separator, and therefore would not include it in a cache key of an unkeyed parameter.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/24297"}, {"title": "Mailing List", "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RSLQD5CCM75IZGAMBDGUZEATYU5YSGJ7/"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpujan2022.html"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202104-04"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2021-3426/pkg:pypi/python@2.7.16", "id": "CVE-2021-3426", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-3426"}, "ratings": [{"score": 5.7, "severity": "medium", "method": "CVSSv31"}], "cwes": [22], "description": "There's a flaw in Python 3's pydoc. A local or adjacent attacker who discovers or is able to convince another local or adjacent user to start a pydoc server could access the server and use it to disclose sensitive information belonging to the other user that they would not normally be able to access. The highest risk of this flaw is to data confidentiality. This flaw affects Python versions before 3.8.9, Python versions before 3.9.3 and Python versions before 3.10.0a7.", "recommendation": "Update to 2.7.18 or later", "advisories": [{"title": "Mailing List", "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VPX7Y5GQDNB4FJTREWONGC4ZSVH7TGHF/"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpujan2022.html"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202104-04"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.18", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2021-3733/pkg:pypi/python@2.7.16", "id": "CVE-2021-3733", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2021-3733"}, "ratings": [{"score": 6.5, "severity": "medium", "method": "CVSSv31"}], "cwes": [400], "description": "There's a flaw in urllib's AbstractBasicAuthHandler class. An attacker who controls a malicious HTTP server that an HTTP client (such as web browser) connects to, could trigger a Regular Expression Denial of Service (ReDOS) during an authentication request with a specially crafted payload that is sent by the server to the client. The greatest threat that this flaw poses is to application availability.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "vendor", "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1995234"}, {"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/24391"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2020-8492/pkg:pypi/python@2.7.16", "id": "CVE-2020-8492", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2020-8492"}, "ratings": [{"score": 6.5, "severity": "medium", "method": "CVSSv31"}], "cwes": [400], "description": "Python 2.7 through 2.7.17, 3.5 through 3.5.9, 3.6 through 3.6.10, 3.7 through 3.7.6, and 3.8 through 3.8.1 allows an HTTP server to conduct Regular Expression Denial of Service (ReDoS) attacks against a client because of urllib.request.AbstractBasicAuthHandler catastrophic backtracking.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/18284"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00011.html"}, {"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00011.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4333-2/"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202005-09"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2019-10160/pkg:pypi/python@2.7.16", "id": "CVE-2019-10160", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-10160"}, "ratings": [{"score": 9.8, "severity": "critical", "method": "CVSSv31"}], "cwes": [522], "description": "A security regression of CVE-2019-9636 was discovered in python since commit d537ab0ff9767ef024f26246899728f0116b1ec3 affecting versions 2.7, 3.5, 3.6, 3.7 and from v3.8.0a4 through v3.8.0b1, which still allows an attacker to exploit CVE-2019-9636 by abusing the user and password parts of a URL. When an application parses user-supplied URLs to store cookies, authentication credentials, or other kind of information, it is possible for an attacker to provide specially crafted URLs to make the application locate host-related information (e.g. cookies, authentication data) and send them to a different host than where it should, unlike if the URLs had been correctly parsed. The result of an attack may vary based on the application.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2019-15903/pkg:pypi/python@2.7.16", "id": "CVE-2019-15903", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-15903"}, "ratings": [{"score": 7.5, "severity": "high", "method": "CVSSv31"}], "cwes": [125], "description": "In libexpat before 2.2.8, crafted XML input could fool the parser into changing from DTD parsing to document parsing too early; a consecutive call to XML_GetCurrentLineNumber (or XML_GetCurrentColumnNumber) then resulted in a heap-based buffer over-read.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/libexpat/libexpat/pull/318"}, {"title": "GitHub Issue", "url": "https://github.com/libexpat/libexpat/issues/342"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4335-1/"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpuoct2020.html"}, {"title": "exploit", "url": "http://seclists.org/fulldisclosure/2019/Dec/30"}, {"title": "Mailing List", "url": "http://lists.opensuse.org/opensuse-security-announce/2020-01/msg00040.html"}, {"title": "Debian Security", "url": "https://www.debian.org/security/2019/dsa-4571"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/201911-08"}, {"title": "Apple Security", "url": "https://support.apple.com/kb/HT210794"}], "analysis": {"state": "exploitable", "detail": "See http://seclists.org/fulldisclosure/2019/Dec/30"}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed\\nKnown Exploits"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-16056/pkg:pypi/python@2.7.16", "id": "CVE-2019-16056", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-16056"}, "ratings": [{"score": 7.5, "severity": "high", "method": "CVSSv31"}], "cwes": [], "description": "An issue was discovered in Python through 2.7.16, 3.x through 3.5.7, 3.6.x through 3.6.9, and 3.7.x through 3.7.4. The email module wrongly parses email addresses that contain multiple @ characters. An application that uses the email module and implements some kind of checks on the From/To headers of a message could be tricked into accepting an email address that should be denied. An attack may be the same as in CVE-2019-11340; however, this CVE applies to Python more generally.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4151-2/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-16935/pkg:pypi/python@2.7.16", "id": "CVE-2019-16935", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-16935"}, "ratings": [{"score": 6.1, "severity": "medium", "method": "CVSSv31"}], "cwes": [79], "description": "The documentation XML-RPC server in Python through 2.7.16, 3.x through 3.6.9, and 3.7.x through 3.7.4 has XSS via the server_title field. This occurs in Lib/DocXMLRPCServer.py in Python 2.x, and in Lib/xmlrpc/server.py in Python 3.x. If set_server_title is called with untrusted input, arbitrary JavaScript can be delivered to clients that visit the http URL for this server.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/16373"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4151-2/"}, {"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2021/04/msg00015.html"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2021/04/msg00015.html"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-18348/pkg:pypi/python@2.7.16", "id": "CVE-2019-18348", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-18348"}, "ratings": [{"score": 6.1, "severity": "medium", "method": "CVSSv31"}], "cwes": [74], "description": "An issue was discovered in urllib2 in Python 2.x through 2.7.17 and urllib in Python 3.x through 3.8.0. CRLF injection is possible if the attacker controls a url parameter, as demonstrated by the first argument to urllib.request.urlopen with \\r\\n (specifically in the host component of a URL) followed by an HTTP header. This is similar to the CVE-2019-9740 query string issue and the CVE-2019-9947 path string issue. (This is not exploitable when glibc has CVE-2016-10739 fixed.). This is fixed in: v2.7.18, v2.7.18rc1; v3.5.10, v3.5.10rc1; v3.6.11, v3.6.11rc1, v3.6.12; v3.7.8, v3.7.8rc1, v3.7.9; v3.8.3, v3.8.3rc1, v3.8.4, v3.8.4rc1, v3.8.5, v3.8.6, v3.8.6rc1.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpuoct2020.html"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00011.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4333-2/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-3902/pkg:pypi/mercurial@4.8.2", "id": "CVE-2019-3902", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-3902"}, "ratings": [{"score": 5.9, "severity": "medium", "method": "CVSSv31"}], "cwes": [59], "description": "A flaw was found in Mercurial before 4.9. It was possible to use symlinks and subrepositories to defeat Mercurial's path-checking logic and write files outside a repository.", "recommendation": "Update to 4.9 or later", "advisories": [{"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00032.html"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/07/msg00032.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4086-1/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/mercurial@4.8.2", "versions": [{"version": "4.8.2", "status": "affected"}, {"version": "4.9", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "false"}]}, {"bom-ref": "CVE-2019-9636/pkg:pypi/python@2.7.16", "id": "CVE-2019-9636", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-9636"}, "ratings": [{"score": 9.8, "severity": "critical", "method": "CVSSv31"}], "cwes": [], "description": "Python 2.7.x through 2.7.16 and 3.x through 3.7.2 is affected by: Improper Handling of Unicode Encoding (with an incorrect netloc) during NFKC normalization. The impact is: Information disclosure (credentials, cookies, etc. that are cached against a given hostname). The components are: urllib.parse.urlsplit, urllib.parse.urlparse. The attack vector is: A specially crafted URL could be incorrectly parsed to locate cookies or authentication data and send that information to a different host than when parsed correctly. This is fixed in: v2.7.17, v2.7.17rc1, v2.7.18, v2.7.18rc1; v3.5.10, v3.5.10rc1, v3.5.7, v3.5.8, v3.5.8rc1, v3.5.8rc2, v3.5.9; v3.6.10, v3.6.10rc1, v3.6.11, v3.6.11rc1, v3.6.12, v3.6.9, v3.6.9rc1; v3.7.3, v3.7.3rc1, v3.7.4, v3.7.4rc1, v3.7.4rc2, v3.7.5, v3.7.5rc1, v3.7.6, v3.7.6rc1, v3.7.7, v3.7.7rc1, v3.7.8, v3.7.8rc1, v3.7.9.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/12201"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpujul2022.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202003-26"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-9674/pkg:pypi/python@2.7.16", "id": "CVE-2019-9674", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-9674"}, "ratings": [{"score": 7.5, "severity": "high", "method": "CVSSv31"}], "cwes": [400], "description": "Lib/zipfile.py in Python through 3.7.2 allows remote attackers to cause a denial of service (resource consumption) via a ZIP bomb.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "Mailing List", "url": "http://lists.opensuse.org/opensuse-security-announce/2020-05/msg00041.html"}, {"title": "vendor", "url": "https://usn.ubuntu.com/4428-1/"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4428-1/"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-9740/pkg:pypi/python@2.7.16", "id": "CVE-2019-9740", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-9740"}, "ratings": [{"score": 6.1, "severity": "medium", "method": "CVSSv31"}], "cwes": [93], "description": "An issue was discovered in urllib2 in Python 2.x through 2.7.16 and urllib in Python 3.x through 3.7.3. CRLF injection is possible if the attacker controls a url parameter, as demonstrated by the first argument to urllib.request.urlopen with \\r\\n (specifically in the query string after a ? character) followed by an HTTP header or a Redis command. This is fixed in: v2.7.17, v2.7.17rc1, v2.7.18, v2.7.18rc1; v3.5.10, v3.5.10rc1, v3.5.8, v3.5.8rc1, v3.5.8rc2, v3.5.9; v3.6.10, v3.6.10rc1, v3.6.11, v3.6.11rc1, v3.6.12, v3.6.9, v3.6.9rc1; v3.7.4, v3.7.4rc1, v3.7.4rc2, v3.7.5, v3.7.5rc1, v3.7.6, v3.7.6rc1, v3.7.7, v3.7.7rc1, v3.7.8, v3.7.8rc1, v3.7.9.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "Mailing List", "url": "http://www.openwall.com/lists/oss-security/2021/02/04/2"}, {"title": "vendor", "url": "https://www.oracle.com/security-alerts/cpujul2022.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}, {"title": "exploit", "url": "https://seclists.org/bugtraq/2019/Oct/29"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202003-26"}], "analysis": {"state": "exploitable", "detail": "See https://seclists.org/bugtraq/2019/Oct/29"}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed\\nKnown Exploits"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-9947/pkg:pypi/python@2.7.16", "id": "CVE-2019-9947", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-9947"}, "ratings": [{"score": 6.1, "severity": "medium", "method": "CVSSv31"}], "cwes": [93], "description": "An issue was discovered in urllib2 in Python 2.x through 2.7.16 and urllib in Python 3.x through 3.7.3. CRLF injection is possible if the attacker controls a url parameter, as demonstrated by the first argument to urllib.request.urlopen with \\r\\n (specifically in the path component of a URL that lacks a ? character) followed by an HTTP header or a Redis command. This is similar to the CVE-2019-9740 query string issue. This is fixed in: v2.7.17, v2.7.17rc1, v2.7.18, v2.7.18rc1; v3.5.10, v3.5.10rc1, v3.5.8, v3.5.8rc1, v3.5.8rc2, v3.5.9; v3.6.10, v3.6.10rc1, v3.6.11, v3.6.11rc1, v3.6.12, v3.6.9, v3.6.9rc1; v3.7.4, v3.7.4rc1, v3.7.4rc2, v3.7.5, v3.7.5rc1, v3.7.6, v3.7.6rc1, v3.7.7, v3.7.7rc1, v3.7.8, v3.7.8rc1, v3.7.9.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "Mailing List", "url": "http://www.openwall.com/lists/oss-security/2021/02/04/2"}, {"title": "vendor", "url": "http://www.openwall.com/lists/oss-security/2021/02/04/2"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202003-26"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2019-9948/pkg:pypi/python@2.7.16", "id": "CVE-2019-9948", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2019-9948"}, "ratings": [{"score": 9.1, "severity": "critical", "method": "CVSSv31"}], "cwes": [22], "description": "urllib in Python 2.x through 2.7.16 supports the local_file: scheme, which makes it easier for remote attackers to bypass protection mechanisms that blacklist file: URIs, as demonstrated by triggering a urllib.urlopen('local_file:///etc/passwd') call.", "recommendation": "Update to 2.7.17 or later", "advisories": [{"title": "GitHub PR", "url": "https://github.com/python/cpython/pull/11842"}, {"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}, {"title": "exploit", "url": "https://seclists.org/bugtraq/2019/Oct/29"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202003-26"}], "analysis": {"state": "exploitable", "detail": "See https://seclists.org/bugtraq/2019/Oct/29"}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "2.7.17", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed\\nKnown Exploits"}, {"name": "depscan:prioritized", "value": "true"}]}, {"bom-ref": "CVE-2018-20852/pkg:pypi/python@2.7.16", "id": "CVE-2018-20852", "source": {"name": "NVD", "url": "https://nvd.nist.gov/vuln/detail/CVE-2018-20852"}, "ratings": [{"score": 5.3, "severity": "medium", "method": "CVSSv31"}], "cwes": [20], "description": "http.cookiejar.DefaultPolicy.domain_return_ok in Lib/http/cookiejar.py in Python before 3.7.3 does not correctly validate the domain: it can be tricked into sending existing cookies to the wrong server. An attacker may abuse this flaw by using a server with a hostname that has another valid hostname as a suffix (e.g., pythonicexample.com to steal cookies for example.com). When a program uses http.cookiejar.DefaultPolicy and tries to do an HTTP connection to an attacker-controlled server, existing cookies can be leaked to the attacker. This affects 2.x through 2.7.16, 3.x before 3.4.10, 3.5.x before 3.5.7, 3.6.x before 3.6.9, and 3.7.x before 3.7.3.", "recommendation": "Update to 3.6.14 or later", "advisories": [{"title": "Mailing List", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "vendor", "url": "https://lists.debian.org/debian-lts-announce/2020/08/msg00034.html"}, {"title": "Ubuntu Security", "url": "https://usn.ubuntu.com/4127-1/"}, {"title": "Gentoo Security", "url": "https://security.gentoo.org/glsa/202003-26"}], "analysis": {}, "affects": [{"ref": "pkg:pypi/python@2.7.16", "versions": [{"version": "2.7.16", "status": "affected"}, {"version": "3.6.14", "status": "unaffected"}]}], "properties": [{"name": "depscan:insights", "value": "Vendor Confirmed"}, {"name": "depscan:prioritized", "value": "true"}]}]}